src/v2i-hub/MobilityOperationPlugin/src/MobilityOperationPlugin.h

```diff
@@ -51,23 +52,29 @@ class MobilityOperationPlugin: public PluginClient {
 
 	void OnStateChange(IvpPluginState state);
 
-	void createPortDrayageJson( PortDrayage_Object &pd_obj, ptree &json_payload);
+	ptree createPortDrayageJson( PortDrayage_Object &pd_obj);
 
-	void createMobilityOperationXml( ptree &mobilityOperationXml, ptree json_payload);
+	ptree createMobilityOperationXml( ptree &json_payload);
 
 	void HandleMobilityOperationMessage(tsm3Message &msg, routeable_message &routeableMsg);
 
-private:
+	PortDrayage_Object retreiveNextAction( std::string action_id );
```

Comment:
I am not sure what the normal header file style is for V2XHub, but it could be good to include the function header comments from the .cpp file for each of the function declarations in the header just to provide someone with context when reading through the file.